includes/admin/class-wc-admin-addons.php

```diff
@@ -127,534 +133,111 @@ public static function get_sections() {
+	 * Handles the outputting of featured page
 	 *
-	 * @return array
+	 * @param array $blocks Featured page's blocks.
 	 */
-	public static function get_section_data( $section_id ) {
-		$section      = self::get_section( $section_id );
-		$section_data = '';
-
-		if ( ! empty( $section->endpoint ) ) {
-			$section_data = get_transient( 'wc_addons_section_' . $section_id );
-			if ( false === $section_data ) {
-				$raw_section = wp_safe_remote_get( esc_url_raw( $section->endpoint ), array( 'user-agent' => 'WooCommerce Addons Page' ) );
-
-				if ( ! is_wp_error( $raw_section ) ) {
-					$section_data = json_decode( wp_remote_retrieve_body( $raw_section ) );
-
-					if ( ! empty( $section_data->products ) ) {
-						set_transient( 'wc_addons_section_' . $section_id, $section_data, WEEK_IN_SECONDS );
-					}
-				}
-			}
-		}
-
-		return apply_filters( 'woocommerce_addons_section_data', $section_data->products, $section_id );
-	}
-
-	/**
-	 * Handles the outputting of a contextually aware Storefront link (points to child themes if Storefront is already active).
-	 */
-	public static function output_storefront_button() {
-		$template   = get_option( 'template' );
-		$stylesheet = get_option( 'stylesheet' );
-
-		if ( 'storefront' === $template ) {
-			if ( 'storefront' === $stylesheet ) {
-				$url         = 'https://woocommerce.com/product-category/themes/storefront-child-theme-themes/';
-				$text        = __( 'Need a fresh look? Try Storefront child themes', 'woocommerce' );
-				$utm_content = 'nostorefrontchildtheme';
-			} else {
-				$url         = 'https://woocommerce.com/product-category/themes/storefront-child-theme-themes/';
-				$text        = __( 'View more Storefront child themes', 'woocommerce' );
-				$utm_content = 'hasstorefrontchildtheme';
+	private static function output_featured( $blocks ) {
+		foreach ( $blocks as $block ) {
+			$block_type = $block->type ?? null;
+			switch ( $block_type ) {
+				case 'group':
+					self::output_group( $block );
+					break;
+				case 'banner':
+					self::output_banner( $block );
+					break;
 			}
-		} else {
-			$url         = 'https://woocommerce.com/storefront/';
-			$text        = __( 'Need a theme? Try Storefront', 'woocommerce' );
-			$utm_content = 'nostorefront';
 		}
-
-		$url = add_query_arg(
-			array(
-				'utm_source'   => 'addons',
-				'utm_medium'   => 'product',
-				'utm_campaign' => 'woocommerceplugin',
-				'utm_content'  => $utm_content,
-			),
-			$url
-		);
-
-		echo '<a href="' . esc_url( $url ) . '" class="add-new-h2">' . esc_html( $text ) . '</a>' . "\n";
 	}
 
 	/**
-	 * Handles the outputting of a banner block.
+	 * Render a group block including products
+	 *
+	 * @param mixed $block Block of the page for rendering.
 	 *
-	 * @param object $block Banner data.
+	 * @return void
 	 */
-	public static function output_banner_block( $block ) {
+	private static function output_group( $block ) {
+		$capacity           = $block->capacity ?? 3;
```

Comment:
Minor thing (PHPCS):

```suggestion
		$capacity             = $block->capacity ?? 3;
```

Reply:
Applied in https://github.com/woocommerce/woocommerce/pull/30900/commits/f54d7b0991520208fa826f1147e29d6fa04e27f5, thanks!